src/cmap/connection_pool.ts

```diff
@@ -530,6 +537,12 @@ function createConnection(pool: ConnectionPool, callback?: Callback<Connection>)
   };
 
   pool[kPending]++;
+  // This is our version of a "virtual" no-I/O connection as the spec requires
+  pool.emit(
+    ConnectionPool.CONNECTION_CREATED,
+    new ConnectionCreatedEvent(pool, { id: connectOptions.id } as Connection)
```

Comment:
From tech chat: We can change the constructor to accept `{id: number|string}` and it will still accept a connection. It's internal so we're fine to do this.

Reply:
done